bin/node/src/config.rs

```diff
@@ -1,54 +1,92 @@
 use miden_node_block_producer::config::BlockProducerConfig;
 use miden_node_rpc::config::RpcConfig;
 use miden_node_store::config::StoreConfig;
+use miden_node_utils::config::Endpoint;
 use serde::{Deserialize, Serialize};
 
 /// Node top-level configuration.
-#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
+#[derive(Clone, Default, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
 pub struct NodeConfig {
-    pub block_producer: Option<BlockProducerConfig>,
-    pub rpc: Option<RpcConfig>,
-    pub store: Option<StoreConfig>,
+    block_producer: NormalizedBlockProducerConfig,
+    rpc: NormalizedRpcConfig,
+    store: StoreConfig,
+}
+
+/// A specialized variant of [RpcConfig] with redundant fields within [NodeConfig] removed.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
+struct NormalizedRpcConfig {
+    endpoint: Endpoint,
+}
+
+/// A specialized variant of [BlockProducerConfig] with redundant fields within [NodeConfig] removed.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
+struct NormalizedBlockProducerConfig {
+    endpoint: Endpoint,
+    verify_tx_proofs: bool,
+}
+
+impl Default for NormalizedRpcConfig {
+    fn default() -> Self {
+        // Ensure we stay in sync with the original defaults.
+        Self { endpoint: RpcConfig::default().endpoint }
+    }
 }
 
-impl Default for NodeConfig {
+impl Default for NormalizedBlockProducerConfig {
     fn default() -> Self {
+        // Ensure we stay in sync with the original defaults.
+        let original = BlockProducerConfig::default();
```

Comment:
I wonder if we should destructure the result here - e.g., something like this:
```rust
let BlockProducerConfig { endpoint, verify_tx_proof, ... } = BlockProducerConfig::default();
```
This way, if `BlockProducerConfig` changes, we'd get a compile-time error here.